### PR TITLE
Refactor of importers and implementation of "populators".

### PR DESCRIPTION
**Pull request by adam-vessey**
If using `islandora_plupload`, requires https://github.com/discoverygarden/islandora_plupload/pull/11 to avoid some errors

"Populator" implementations are dependent on https://github.com/discoverygarden/islandora_populator/pull/1

### CODE REVIEW

**Comment by adam-vessey** [comment]
Moving around the exslt stuff sure looks ugly in the diff. :P

**Comment by jordandukart** [comment]
Yep.

File: modules/doi/modules/doi_importer/doi_importer.install [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * @file
+ * Install and update hook implementations.
+ */
+
+/**
+ * Deal with refactor of functions to islandora_doi.
+ */
+function doi_importer_update_7100() {
+  module_enable('islandora_doi');
```
Comment (jordandukart) [inline modules/doi/modules/doi_importer/doi_importer.install]: `array` plx?

File: modules/endnotexml/modules/endnotexml_importer/endnotexml_importer.install [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+/**
+ * @file
+ * Installation and update hook implementations.
+ */
+
+/**
+ * Enable new dependency (which should be present...).
+ */
+function endnotexml_importer_update_7100() {
+  module_enable('islandora_endnotexml');
```
Comment (jordandukart) [inline modules/endnotexml/modules/endnotexml_importer/endnotexml_importer.install]: `array` plx?

File: modules/pmid/modules/pmid_importer/pmid_importer.install [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+/**
+ * @file
+ * Installation and update hook implementations.
+ */
+
+/**
+ * Add in new "islandora_pmid" dependency.
+ */
+function pmid_importer_update_7100() {
+  module_enable('islandora_pmid');
```
Comment (jordandukart) [inline modules/pmid/modules/pmid_importer/pmid_importer.install]: `array` :octopus:

**Pull request merged**